Framework/API/src/FileFinder.cpp

```diff
@@ -570,17 +510,50 @@ FileFinderImpl::findRun(const std::string &hintstr,
   return "";
 }
 
+/**
+ * Given a set of already determined extensions and new extensions,
+ * create a set of all extensions.
+ * If not in an extension-is-case-sensitive environment, only add the
+ * lower case OR upper case version of the extension
+ * @param extensionsToAdd :: a vector of extensions to add
+ * @param uniqueExts :: a vector of currently included extensions
+ */
+void FileFinderImpl::getUniqueExtensions(
+    const std::vector<std::string> &extensionsToAdd,
+    std::vector<std::string> &uniqueExts) const {
+  const bool isCaseSensitive = getCaseSensitive();
+  for (const auto &cit : extensionsToAdd) {
+    std::string transformed(cit);
+    if (!isCaseSensitive) {
+      std::transform(cit.begin(), cit.end(), transformed.begin(), tolower);
+    }
+    const auto searchItr =
+        std::find(uniqueExts.begin(), uniqueExts.end(), transformed);
+    if (searchItr == uniqueExts.end()) {
+      uniqueExts.push_back(transformed);
+    }
+  }
+}
+
 /**
  * Find a list of files file given a hint. Calls findRun internally.
  * @param hintstr :: Comma separated list of hints to findRun method.
  *  Can also include ranges of runs, e.g. 123-135 or equivalently 123-35.
  *  Only the beginning of a range can contain an instrument name.
+ * @param exts :: Vector of allowed file extensions. Optional.
+ *                If provided, this provides the only extensions searched for.
+ *                If not provided, facility extensions used.
+ * @param useExtsOnly :: Optional bool. If it's true (and exts is not empty),
+                           search the for the file using exts only.
+                           If it's false, use exts AND facility extensions.
  * @return A vector of full paths or empty vector
  * @throw std::invalid_argument if the argument is malformed
  * @throw Exception::NotFoundError if a file could not be found
  */
 std::vector<std::string>
-FileFinderImpl::findRuns(const std::string &hintstr) const {
+FileFinderImpl::findRuns(const std::string &hintstr,
```

Comment:
Could we leave a function override which still uses the old inputs so as not to break old code?

Reply:
I'm not really convinced we would need to overload `findRuns` as using the default values of an empty vector and `false` results in exactly the same behaviour as before, so old code which calls `findRuns` with hinstr only will work the same. As `findRuns` calls `findRun` the only real difference we now have with the old code is that it's calling `findRun` and explicitly passing in an empty vector rather than using the default empty vector in `findRun`.

I think overloading the function to have the old `findRuns` will either require us to just call something like `findRuns(hinstr, {}, false)`, which is a less clean way of what we have here, or require duplicated code. Perhaps I'm misunderstanding what you have in mind though

Reply:
Yes I think you are right here, provided old usage still work all is good.